core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java

```diff
@@ -176,13 +176,8 @@ public final synchronized RunT newBuild() throws IOException {
             builds.put(lastBuild);
             lastBuild.getPreviousBuild(); // JENKINS-20662: create connection to previous build
             return lastBuild;
-        } catch (InstantiationException e) {
-            throw new Error(e);
-        } catch (IllegalAccessException e) {
-            throw new Error(e);
-        } catch (InvocationTargetException e) {
```

Comment:
Isn't a change in behaviour? Or is it done on purpose?
`handleInvocationTargetException(e);` basically unwraps the target exception, and now it's not the case anymore.